### DISCUSSION TITLE
v0.21.0

### DISCUSSION THREAD

**Discussion by erogol** [coqui-ai/TTS#3301] [Versions & Releases]
## What's Changed
* Remove duplicate/unused code by @eginhard in https://github.com/coqui-ai/TTS/pull/3243
* Making the Model Manager's Progress bar statically accessible via the class. by @FlorianEagox in https://github.com/coqui-ai/TTS/pull/3297
* More informative error for wrong --language argument by @eginhard in https://github.com/coqui-ai/TTS/pull/3294
* Don't pass quotes to espeak by @eginhard in https://github.com/coqui-ai/TTS/pull/3286
* Fix tts_with_vc by @eginhard in https://github.com/coqui-ai/TTS/pull/3275
* Misjudgment of `is_multi_lingual` When Loading Multilingual Model via `model_path` by @TITC in https://github.com/coqui-ai/TTS/pull/3273
* Introducing Development Dockerfile by @Kaszanas in https://github.com/coqui-ai/TTS/pull/3263
* update deepspeed version by @WeberJulian in https://github.com/coqui-ai/TTS/pull/3281

## New Contributors
* @FlorianEagox made their first contribution in https://github.com/coqui-ai/TTS/pull/3297
* @TITC made their first contribution in https://github.com/coqui-ai/TTS/pull/3273
* @Kaszanas made their first contribution in https://github.com/coqui-ai/TTS/pull/3263

**Full Changelog**: https://github.com/coqui-ai/TTS/compare/v0.20.6...v0.21.0

<hr /><em>This discussion was created from the release <a href='https://github.com/coqui-ai/TTS/releases/tag/v0.21.0'>v0.21.0</a>.</em>

**Comment by erew123** [comment]
Since updating to the new tts_models/multilingual/multi-dataset/xtts_v2 model (today) and also updating from TTS 0.20.6 to 0.21.0, while using a sample UK English voice, it now comes out sounding like a quite strange American and no longer like the sample wav file, which it did on the prior version of xtts_v2 (no matter how many samples I create).

Or perhaps I should say, it starts out kind of ok for the first 10 seconds and then starts to trend towards sounding American, which it didnt before.

I appreciate its all a work, in progress and not sure if this is the right place to feed back about the models.

EDIT - Because its not showing clearly on this message chain. There are some **here are some solutions** for "**It keeps downloading"** and **"I want to use the old model"**. https://github.com/coqui-ai/TTS/discussions/3301#discussioncomment-7663904

Thanks

**Reply by Tessory** [reply]
I updated only model, and still didn't update TTS. Also got strange English accent(American or not I can't understand) on my non-english language. I rollback old model, and there all fine.

**Reply by erew123** [reply]
FYI, it kept forcing me to re-download the model every time the model loaded into RAM, until I updated TTS! One to watch out for!

But, that's good to know, its the model and not TTS. 

Found the old model https://huggingface.co/coqui/XTTS-v2/tree/v2.0.2

I just need to find a way to force it to stop updating!! (if you know one let me know)

**Reply by Tessory** [reply]
I turn off internet for run TTS without update model. You can find me on discord.

**Reply by erew123** [reply]
Thanks! I'm literally about to run out the door, as I have meeting. May hunt you out later on Discord! Thanks for confirming its not me going crazy and that it sounds umm, very different/not quite right. Firewalling python it is I guess :)

**Reply by carinae** [reply]
My nice French voice now turn like a Russian babushka trying to speak French with strange sound on the sentence... Thank for the link to get the old model.

**Reply by erew123** [reply]
Just had time to look at this again. 

I believe you DO have to perform a **pip install --upgrade tts** in your python environment to stop it continuously trying to download a model. This will take you to the version this post is about and it **may** still download the model just the 1x more time  (let it do this, if it does!). If you didnt PIP your install, then I would just update to the latest version of TTS, however you did it before.

**As for the sound quality and reverting back to the 2.0.2 model**.  There is Option A and B

**Option A**
This worked for me, **after** TTS was upgraded:

It looks as if you dump the **model.pth** and **vocab.json** from the 2.0.2 model https://huggingface.co/coqui/XTTS-v2/tree/v2.0.2 over the top of the 2.0.3 mode. In windows this is 
`C:\Users\ {USER} \AppData\Local\tts\tts_models--multilingual--multi-dataset--xtts_v2` 

Linux probably is
`/home/${USER}/.local/share/tts/tts_models--multilingual--multi-dataset--xtts_v2`

 **ONLY THOSE 2 FILES model.pth and vocab.json, Dont touch any of the other files in there**

Then it starts to generate audio like the previous version did (without the strange accent) and it doesn't demand you download the model each time! :)

**Option B**
You can use a custom model path to point to your downloaded model https://github.com/coqui-ai/TTS/discussions/3301#discussioncomment-7664165

**Though I think the Devs need to know that the new model is sounding very strange, based on 5 people other than myself that I've seen so far.** It might be nice if there was a "don't update" option within the TTS engine (cant find one if there is one).

**Reply by erogol** [reply]
We are discussing supporting earlier versions. Until we come to a conclusion... 

You can always download the previous model and then manually load it. 

You can browse the branches and find the version you need -> https://huggingface.co/coqui/XTTS-v2

**Reply by erogol** [reply]
You can also provide your voices so that we can add those to the training, then you never lose them again plus you would get the best possible output. If you are interested ping me on Discord with a DM.

**Reply by carinae** [reply]
Thank you erew123

The upgrade of tts remove the dowloading of the new model, then I replaced the in new model path, model.pth and vocab.json by the old one.
It's working again. (with the issue of the pronunciation of the point, but can be managed by replacing by ',')

I shared the voice on reddit (it's from a free Lexibook based on J.Verne Book) , so the voice I'm using most

[french voice used](https://drive.proton.me/urls/VD9ZQSPW6G#D6T8N56cHTn3)

[Example -> answer to question about constitution of atom ](https://drive.proton.me/urls/252QSFXQS0#tLXSZgucCSZo)

Thank you for you job Erogol, coqui-ai is the best open source result for the French accent (with model 2.0.2 :) )

**Reply by dancodedev** [reply]
I think the problem with the points only happens when you use tts through cli. I also had that problem, making the inference like this code. I don't have the problem with the points.

```
from TTS.api import TTS
tts = TTS("tts_models/multilingual/multi-dataset/xtts_v2", gpu=True)

tts.tts_to_file(
    text="It took me quite a long time to develop a voice, and now that I have it I'm not going to be silent.",
    file_path="output.wav",
    speaker_wav="/path/to/target/speaker.wav",
    language="en"
)
```

**Reply by Tessory** [reply]
How in that kind code use a custom model path?
where here to add path:
tts = TTS("tts_models/multilingual/multi-dataset/xtts_v2", gpu=True)
?

**Comment by erew123** [comment]
> You can always download the previous model and then manually load it.

Do you mean using the **--model_path** and pointing it your downloaded files?  

Id just like to say that the previous version 2.0.2 sounded pretty good, close to the voice that I was sampling. And this latest build 2.0.3, sounds at best 60% as good with very strange accents and it degrades after 15-20 seconds into generation.

I'm absolutely not having a go! Don't get me wrong. I really love what you guys are doing and think its fantastic! But I do want to give some honest feedback :)

Thanks!

**Reply by erogol** [reply]
I don't know how you use TTS. But you can check the docs. It shows you to load models manually in different scenarios. 

Sure thanks for the feedback. Noted. You can always keep using v2.0.2 for your voice until you are satisfied with any new models.

**Reply by erew123** [reply]
@erogol Thanks! 

Just to confirm to others, you can use a custom model path as follows:

`tts --model_path c:/xttsv2_2.0.3/ --config_path c:/xttsv2_2.0.3/config.json --text "This is a test" --speaker_wav c:\AI\Female.wav --language_idx en --use_cuda true`

I copied the entire 2.0.3 model files to a folder I called **xttsv2_2.0.3** (but it could be any of the models you want to use)

From there you need to specify the **--model_path** though you dont add model.pth. So I added **--model_path c:/xttsv2_2.0.3/**

You also need to specify **--config_path** though you DO list the config file name **--config_path c:/xttsv2_2.0.3/config.json**

Obviously, you can do the same for any other model versions, on any path of your choosing.

**Comment by allenhs** [comment]
I have the same issue with the 2.0.3 model. All the voices I created (American voices, started to sound British or have weird accents and like others have said they would degrade as the audio would go on. They were near perfect on 2.0.2. I'm happy for now that I can downgrade to 2.0.2 and keep using that.

**Comment by ElhamAhmedian** [comment]
Keep getting:
"tts_models/multilingual/multi-dataset/xtts_v2 is already downloaded however it has been changed. Redownloading it..."
No matter what I do.

**Reply by erew123** [reply]
Your TTS isnt updated would be my guess. In your Python environment, at a terminal you can do a **pip list** and see what version of TTS you are using. It should be 0.21.1. If its not then the **pip install --upgrade tts**

**Reply by carinae** [reply]
Upgrading TTS worked for me, the model keep the same (previous 2.0.2) and not redownloading it...

Attempting uninstall: tts
    Found existing installation: TTS 0.20.6
    Uninstalling TTS-0.20.6:
      Successfully uninstalled TTS-0.20.6
Successfully installed tts-0.21.1

As erew123 told you, you maybe need to activate the environement where you run coqui-ai...

**Comment by ElhamAhmedian** [comment]
```

[XTTS] Loading XTTS...
 > tts_models/multilingual/multi-dataset/xtts_v2 is already downloaded.
 > tts_models/multilingual/multi-dataset/xtts_v2 is already downloaded however it has been changed. Redownloading it...
 > Downloading model to C:\Users\elham\AppData\Local\tts\tts_models--multilingual--multi-dataset--xtts_v2
 10%|██████████▋                                                                                                   | 182M/1.87G [02:21<15:29, 1.81MiB/s
```
It is the latest version of TTS, and  I am not using the command line, its directly from python.

**Reply by erew123** [reply]
If your python environment is on that version of the TTS engine (not the model that its downloading, the engine e.g. TTS.exe etc) then I don't know what else it could be. I only observed the behaviour of re-downloading the model, when the TTS engine hadn't been updated to the current build. If you have confirmed with **pip list** that you have the current TTS 0.21.1 then this will be something else... maybe the model hash file is funny. I'm not a developer here so maybe they can help you

This is what id expect you would see on the **pip list** at your command prompt.
![image](https://github.com/coqui-ai/TTS/assets/35898566/833de81d-3113-48fc-b617-d0bcf68219d0)

**Reply by marioguima** [reply]
See it "Model xxts_v2 always downloads (solution)"
[https://github.com/coqui-ai/TTS/discussions/3304](https://github.com/coqui-ai/TTS/discussions/3304)

**Comment by marioguima** [comment]
TTS is truly amazing, and I know there are many updates to come.
However, I also needed to go back to version 2.0.2 due to the mixed accent in the pronunciation of the Brazilian Portuguese language.

**Comment by Qualzz** [comment]
I confirm using 2.0.2 lead to waaaay better results, the new model kinda sounds like someone non native tries to speak the language. 
For exemple with french, it sounds like a bad russian imitation. 
Thank you for your work nonetheless and I understand it's hard to evaluate a model especially when working on so much different languages !
